### DISCUSSION TITLE
Persistent data between sessions in dev mode

### DISCUSSION THREAD

**Discussion by SpeedyCraftah** [remix-run/remix#2086] [Q&A] [answered]
So I see that in development mode, data does not persist at all like it would in production mode.
An example would be:
```tsx
// data.ts
export let count = 0;

// page.tsx
import { count } from "./data";

export function loader() {
    count++;
    console.log(count);
}
```

`console.log(count);` will always return 1 in developer mode (this is without any code edits or rebuilds) while in production mode it persists as expected and increments.

I know this is probably intended behaviour since developer mode is supposed to be "cacheless" but is there a way around this (aka persist data between development mode sessions)? Reconnecting to the database on every request makes it very slow and annoying to deal with. You also cannot cache any user sessions (unless you can with the built-in session structure but I find it very confusing and over-complicated for my use case with all due respect).

Thank you!

**Comment by sergiodxa** [comment, marked as answer]
You can store in `global` anything that you want to persists.

**Reply by SpeedyCraftah** [reply]
Yeah that works great, thanks. Would be great if there was a better way of doing this. Possible feature suggestion?

**Reply by cmd-johnson** [reply]
@SpeedyCraftah This should be covered by #1818
And there's even a PR that seems to address this: #2046
Remix would still reset your counter every time you make changes to your code, but at least not between individual requests.
